cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/autocomplete-config.js

```diff
@@ -21,11 +26,20 @@ function enableAutocomplete(config) {
                 panel: "algolia-autocomplete suggestions p-2 d-block",
             },
             placeholder: algoliaData.strings.placeholder,
-            getSources(query) {
-                return getSourcesArray(config);
+            getSources() {
+                const trendingItemsPromise = getTrendingItemsArray();
+
+                // Immediately return other sources that do not depend on trending items
```

Comment:
This comment doesn't seem accurate, I see below that we wait for the `trendingItemsPromise` before returning the `sources`.
Did you try to my suggestion here? https://github.com/algolia/algoliasearch-sfcc-b2c/pull/148#discussion_r1516190323
It would make everything more clean.

Reply:
Hello @sbellone ,
sorry but I believe I didnt understand this comment, can you please clarify it a bit more?